pallets/parachain-staking/src/lib.rs

```diff
@@ -2333,18 +2335,46 @@ pub mod pallet {
 				});
 			};
 		}
-	}
 
-	/// Add reward points to block authors:
-	/// * 20 points to the block producer for producing a block in the chain
-	impl<T: Config> Pallet<T> {
+		/// Add reward points to block authors:
+		/// * 20 points to the block producer for producing a block in the chain
 		fn award_points_to_block_author() {
 			let author = T::BlockAuthor::get();
 			let now = <Round<T>>::get().current;
 			let score_plus_20 = <AwardedPts<T>>::get(now, &author).saturating_add(20);
 			<AwardedPts<T>>::insert(now, author, score_plus_20);
 			<Points<T>>::mutate(now, |x| *x = x.saturating_add(20));
 		}
+
+		/// Marks collators as inactive for the previous round if they received zero awarded points.
+		pub fn mark_collators_as_inactive(cur: RoundIndex) -> Weight {
+			// This function is called after round index increment,
+			// We don't need to saturate here because the genesis round is 1.
+			let prev = cur - 1;
+
+			for (account, _) in <AtStake<T>>::iter_prefix(prev) {
```

Comment:
This currently does not impact the implementation, but it was an extrinsic, a possible improvement would be to wrap the loop inside `if <WasInactive<T>>::iter_prefix(prev).next().is_none()`.